### PR TITLE
[LOOP-928] Enforce deliverable increments on watch

### PR DESCRIPTION
**Pull request by mpangburn**
- Plumbing to support transfer of supported bolus volumes
- Transfer supported bolus volumes when the PumpManager changes
- Use "smart" formatting for the number of decimal places shown during bolus input on watch (the `deltaScale(boundedBy:)` extension method is my own, taken from my work on FractionalQuantityPicker)

### CODE REVIEW

File: Loop/Managers/WatchDataManager.swift [hunk 1]
```diff
@@ -84,6 +86,32 @@ final class WatchDataManager: NSObject {
...
+            session.activate()
+            return
+        }
+
+        guard volumes != lastSentBolusVolumes else {
+            log.default("Skipping bolus volumes transfer due to no changes")
+            return
+        }
+
+        lastSentBolusVolumes = volumes
```
Comment (ps2) [inline Loop/Managers/WatchDataManager.swift:109]: If sending fails, this will prevent future attempts.

**Review by ps2 — COMMENTED** [review]
I think there is an issue with the logic for recording last sent volumes where it is set even if the transfer fails.

**Comment by mpangburn** [comment]
@ps2 I've added logic to `func session(_ session: WCSession, didFinish userInfoTransfer: WCSessionUserInfoTransfer, error: Error?)` to handle this case. Can you review the latest changes?

**Review by darinkrauss — APPROVED** [review]
LGTM! 🦸‍♂️

**Review by ps2 — APPROVED** [review]
LGTM!

**Pull request merged**